README.md

```diff
@@ -246,6 +246,35 @@ Comlink does provide TypeScript types. When you `expose()` something of type `T`
 
 Comlink works with Node’s [`worker_threads`][worker_threads] module. Take a look at the example in the `docs` folder.
 
+### TypeScript and Node
+
+To use Comlink in a TypeScript project running on Node you have to:
+
+1. Install [@types/node](https://www.npmjs.com/package/@types/node) to import the [`worker_threads`][worker_threads] module
+2. Set `"type": "module"` in your `package.json` to use ESM instead of CommonJS
+3. Set `"module": "ESNext"` or `"ES2015"` in your `tsconfig.json` to emit ESM syntax
+4. Use `// @ts-ignore` to import code from files with `.mjs` extensions as long as TypeScript does not [support mjs natively](https://github.com/microsoft/TypeScript/issues/18442)
+5. Run your TypeScript code with an ESM loader (like [ts-node](https://github.com/TypeStrong/ts-node)): `node --loader ts-node/esm ./main.ts`
+
+```ts
+import { Worker } from "worker_threads";
+import { wrap } from "comlink";
+// @ts-ignore
+import nodeEndpoint from "comlink/dist/esm/node-adapter.mjs";
+
+interface WorkerAPI {
+  doMath: () => number;
+}
+
+async function init() {
+  const worker = new Worker("./worker.mjs");
```

Comment:
Does this work with TypeScript worker files, too?

Reply:
... because ideally you wouldn’t define `WorkerAPI` in here but rather import from the worker file. Like, in my head a TypeScript worker file looks like this:

```ts
import {expose} from "comlink";

const api = {
  number() {
    return 42;
  }
};
expose(api);

export type API = typeof api;
```

and in here you do 

```ts
// ...
import type {API} from "./worker";

async function main() {
  const w = new Worker("./worker.ts");  
  const api = wrap<API>(nodeEndpoint(w));
  console.log(await api.number());
}
```

Is that possible? If not, what needs fixing?

Reply:
I agree that it would be better to export the API directly from the worker. Unfortunately, the `Worker` constructor expects a string with a ".js", ".mjs", or ".cjs" extension. Calling `new Worker('./src/worker.ts')` will result in an "ERR_WORKER_UNSUPPORTED_EXTENSION" error. 

The only way, that I have found, to bypass this limitation is to use a JavaScript portal. This portal script is a JavaScript file which requires `ts-node` to execute TypeScript code afterwards:

```js
const {workerData} = require('worker_threads');
require('ts-node').register();
require(workerData.workerPath);
```

But I don't find this convenient enough to list it in the readme. 

Do you have a better idea?

Reply:
@surma how do we continue from here? I think that the current description already provides value. We can sort out loading a pure TypeScript worker in a follow-up PR if you like?